lang/c++/api/NodeImpl.hh

```diff
@@ -113,15 +113,15 @@ protected:
         return docAttribute_.get();
     }
 
-    void doAddLeaf(const NodePtr &newLeaf) override {
+    void doAddLeaf(const NodePtr &newLeaf) final {
```

Comment:
this is semantic in what looks likes a public header, are you sure noone else has subclassed this class, and might have overridden this method?

Reply:
The reason this is final is because this is being invoked from the constructor. Invoking a non-final virtual function from constructor is confusing at the very least.